cmd/operator-sdk/add/api.go

```diff
@@ -21,89 +21,257 @@ import (
+is generated as a 'validation' object.`,
+		Example: `  # Create a new API, under an existing project. This command must be run from the project root directory.
+# Go Example:
+  $ operator-sdk add api --api-version=app.example.com/v1alpha1 --kind=AppService
+
+# Ansible Example
+  $ operator-sdk add api  \
+  --api-version=app.example.com/v1alpha1 \
+  --kind=AppService
+
+# Helm Example:
+  $ operator-sdk add api \
+  --api-version=app.example.com/v1alpha1 \
+  --kind=AppService
+
+  $ operator-sdk add api \
+  --api-version=app.example.com/v1alpha1 \
+  --kind=AppService
+  --helm-chart=myrepo/app
+
+  $ operator-sdk add api \
+  --helm-chart=myrepo/app
+
+  $ operator-sdk add api \
+  --helm-chart=myrepo/app \
+  --helm-chart-version=1.2.3
+
+  $ operator-sdk add api \
+  --helm-chart=app \
+  --helm-chart-repo=https://charts.mycompany.com/
+
+  $ operator-sdk add api \
+  --helm-chart=app \
+  --helm-chart-repo=https://charts.mycompany.com/ \
+  --helm-chart-version=1.2.3
+
+  $ operator-sdk add api \
+  --helm-chart=/path/to/local/chart-directories/app/
+
+  $ operator-sdk add api \
+  --helm-chart=/path/to/local/chart-archives/app-1.2.3.tgz
 `,
 		RunE: apiRun,
 	}
 
-	apiCmd.Flags().StringVar(&apiVersion, "api-version", "",
-		"Kubernetes APIVersion that has a format of $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)")
-	if err := apiCmd.MarkFlagRequired("api-version"); err != nil {
-		log.Fatalf("Failed to mark `api-version` flag for `add api` subcommand as required")
-	}
-	apiCmd.Flags().StringVar(&kind, "kind", "", "Kubernetes resource Kind name. (e.g AppService)")
-	if err := apiCmd.MarkFlagRequired("kind"); err != nil {
-		log.Fatalf("Failed to mark `kind` flag for `add api` subcommand as required")
-	}
-	apiCmd.Flags().BoolVar(&skipGeneration, "skip-generation", false,
-		"Skip generation of deepcopy and OpenAPI code and OpenAPI CRD specs")
-	apiCmd.Flags().StringVar(&crdVersion, "crd-version", gencrd.DefaultCRDVersion,
-		"CRD version to generate")
+	// Initialize flagSet struct with command flags
+	apiFlags.AddTo(apiCmd.Flags())
 
 	return apiCmd
 }
 
 func apiRun(cmd *cobra.Command, args []string) error {
+
 	projutil.MustInProjectRoot()
 
-	// Only Go projects can add apis.
-	if err := projutil.CheckGoProjectCmd(cmd); err != nil {
+	operatorType := projutil.GetOperatorType()
+	if operatorType == projutil.OperatorTypeUnknown {
+		return projutil.ErrUnknownOperatorType{}
+	}
+	// Verify the incoming flags.
+	if err := apiFlags.VerifyCommonFlags(operatorType); err != nil {
+		return err
+	}
+
+	log.Infof("Generating api version %s for kind %s.", apiFlags.APIVersion, apiFlags.Kind)
+
+	switch operatorType {
+	case projutil.OperatorTypeGo:
+		if err := doGoAPIScaffold(); err != nil {
+			return err
+		}
+	case projutil.OperatorTypeAnsible:
+		if err := doAnsibleAPIScaffold(); err != nil {
+			return err
+		}
+	case projutil.OperatorTypeHelm:
+		if err := doHelmAPIScaffold(); err != nil {
+			return err
+		}
+	default:
+	}
```

Comment:
I understand that the default should return `projutil.ErrUnknownOperatorType{}`

Reply:
Honestly `default:` won't do anything. We check the type on line 121 and return the error on line 123. The type does not get changed after that point. So no error should be returned. Nit: we can remove it but I won't hold up the PR for it.